Dockerfile

```diff
@@ -14,7 +14,7 @@ USER cryptol
 WORKDIR /cryptol
 RUN mkdir -p rootfs/usr/local/bin
 WORKDIR /cryptol/rootfs/usr/local/bin
-RUN curl -o solvers.zip -sL "https://github.com/GaloisInc/what4-solvers/releases/download/snapshot-20221212/ubuntu-22.04-bin.zip"
+RUN curl -o solvers.zip -sL "https://github.com/GaloisInc/what4-solvers/releases/download/snapshot-20230711/ubuntu-22.04-X64-bin.zip"
```

Comment:
Maybe add a comment here (and the other Dockerfile) that the zipfile name is based on the CI matrix.os and runner.arch so that future devs know where this name came from.  Perhaps even a contra-comment in the YAML files as well indicating that the BIN_ZIP_FILE is consumed by these two dockerfiles.

Reply:
I can certainly leave a comment about the naming convention here, but the Dockerfiles have nothing to do with `BIN_ZIP_FILE`—that environment variable is only used by `.github/ci.sh`. (Perhaps you meant to suggest adding a comment there?)

Reply:
Yes, I was unclear: it might be helpful to have a comment there that says that the name specified for BIN_ZIP_FILE will be consumed by the dockerfiles, not the BIN_ZIP_FILE variable itself.  My concern is that if someone changes the BIN_ZIP_FILE value in the CI yaml files that they should know they will need to make corresponding changes in the dockerfiles.

Reply:
I've added a commit which adds some pointers next to `SOLVER_PKG_VERSION` and `BIN_ZIP_FILE` in the `ci.yml` file, as well as the lines in each Dockerfile that were inspired by them.